subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ResolveConfigurationRepositoriesBuildOperationIntegrationTest.groovy

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.resolve
+
+import com.google.common.collect.Maps
+import org.gradle.api.internal.artifacts.configurations.ResolveConfigurationDependenciesBuildOperationType
+import org.gradle.integtests.fixtures.AbstractHttpDependencyResolutionTest
+import org.gradle.integtests.fixtures.BuildOperationNotificationsFixture
+import org.gradle.integtests.fixtures.BuildOperationsFixture
+import org.gradle.test.fixtures.plugin.PluginBuilder
+import spock.lang.Unroll
+
+class ResolveConfigurationRepositoriesBuildOperationIntegrationTest extends AbstractHttpDependencyResolutionTest {
```

Comment:
We don't actually need the stability guarantees that are being tested here, so this test is misleading.

We do not need the IDs to be stable. We _only_  the `getRepositoryId()` on component to match one the ID of one of the repos in the details for the associated details of the configuration resolve. However, all of the other logical properties should be stable, because that is how we will determine that they are the same thing (more or less)

I think we should relax the tests in this regard. Or, somehow make it _very_ clear that we do not require IDs to be stable.

Reply:
relaxed the test